### PR TITLE
Add case of connectivity check of mcast interface

### PR DESCRIPTION
**Pull request by chloerh**
- VIRT-294957 - [mcast] Check connectivity of mcast type interface

test result:
 (1/1) type_specific.io-github-autotest-libvirt.virtual_network.connectivity_check.mcast_interface.default: PASS (108.24 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /var/lib/avocado/job-results/job-2023-08-31T23.33-6281078/results.html
JOB TIME   : 109.69 s

### CODE REVIEW

File: libvirt/tests/src/virtual_network/connectivity/connectivity_check_mcast_interface.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+        mac, mac_c = list(map(vm_xml.VMXML.get_first_mac_by_name, vms))
+
+        ip = utils_net.get_guest_ip_addr(session, mac)
+        ip_c = utils_net.get_guest_ip_addr(session_c, mac_c)
+
+        LOG.debug(session.cmd_output('ip a'))
+        LOG.debug(session_c.cmd_output('ip a'))
+
+        # Build omping command
```
Comment (yanqzhan) [inline libvirt/tests/src/virtual_network/connectivity/connectivity_check_mcast_interface.py:61]: need install 'omping' pkg first.

File: libvirt/tests/src/virtual_network/connectivity/connectivity_check_mcast_interface.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+        for vmxml_i in (vmxml, vmxml_c):
+            for iface_i in (iface, iface_m):
+                libvirt.add_vm_device(vmxml_i, iface_i)
+
+        [LOG.debug(f'VMXML of {vm_x}:\n{virsh.dumpxml(vm_x).stdout_text}')
+         for vm_x in vms]
+
+        [vm_i.start() for vm_i in [vm, vm_c]]
+
```
Comment (yanqzhan) [inline libvirt/tests/src/virtual_network/connectivity/connectivity_check_mcast_interface.py:49]: missing this checkpoint:
# netstat -g | grep 230.144
eno1             2      230.144.17.1

File: libvirt/tests/src/virtual_network/connectivity/connectivity_check_mcast_interface.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+        session = vm.wait_for_serial_login()
+        session_c = vm_c.wait_for_serial_login()
+
+        mac, mac_c = list(map(vm_xml.VMXML.get_first_mac_by_name, vms))
```
Comment (yanqzhan) [inline libvirt/tests/src/virtual_network/connectivity/connectivity_check_mcast_interface.py:53]: seems need to get second mac for interface ip, the first one doesn't have ip in vm os.

File: libvirt/tests/src/virtual_network/connectivity/connectivity_check_mcast_interface.py (hunk 1 above)
Comment (chloerh) [reply libvirt/tests/src/virtual_network/connectivity/connectivity_check_mcast_interface.py:68]: Updated

File: libvirt/tests/src/virtual_network/connectivity/connectivity_check_mcast_interface.py (hunk 2 above)
Comment (chloerh) [reply libvirt/tests/src/virtual_network/connectivity/connectivity_check_mcast_interface.py:51]: Updated.

File: libvirt/tests/src/virtual_network/connectivity/connectivity_check_mcast_interface.py (hunk 3 above)
Comment (chloerh) [reply libvirt/tests/src/virtual_network/connectivity/connectivity_check_mcast_interface.py]: Updated to vm.get_address()

**Comment by chloerh** [comment]
Test result:

PASS 1-type_specific.io-github-autotest-libvirt.virtual_network.connectivity_check.mcast_interface.default

**Review by yanqzhan — APPROVED** [review]
LGTM.

**Review by Yingshun — APPROVED** [review]

**Pull request merged**